util/checktr.py

```diff
@@ -0,0 +1,410 @@
+    '''Test all locale files in path and subdirectories'''
+    files = 0
+    filefails = 0
+    filewarns = 0
+    empties = 0
+    tr_files = get_existing_tr_files(path)
+    for file in tr_files:
+        files += 1
+        (errors, warnings, empty_translations, total_strings) = check_tr_file(file, False, toptions)
+        if errors > 0:
+            filefails += 1
+        if warnings > 0:
+            filewarns += 1
+        if empty_translations > 0:
+            empties += 1
+
+    template_files = get_existing_template_files(path)
+    for file in template_files:
+        files += 1
+        (errors, warnings, empty_translations, total_strings) = check_tr_file(file, True, toptions)
+        if errors > 0:
+            filefails += 1
+        if warnings > 0:
+            filewarns += 1
+
+    if files == 0:
+        print("No *.tr or template.txt files found in "+path)
+        return True
+
+    print("------------")
+    print("TEST RESULT:")
+    print("------------")
+    if filefails == 0:
+        print(str(files) + " file(s) checked. Test PASSED!")
+        if filewarns > 0:
+            print(str(filewarns) + " file(s) with warnings.")
+        if toptions["list_empty"] or empties > 0:
+            print(str(empties) + " file(s) with empty translations.")
+        return True
+    else:
+        print(str(files) + " file(s) checked. Test FAILED!")
+        print(str(filefails) + " file(s) with error(s).")
+        if filewarns > 0:
+            print(str(filewarns) + " file(s) with warnings.")
+        if toptions["list_empty"] or empties > 0:
+            print(str(empties) + " file(s) with empty translations.")
+        return False
+
+
+def set_params(tab: list):
+    '''Initialize params from CLI arguments.'''
+    for option in options:
+        for option_name in options[option]:
+            if option_name in tab:
+                params[option] = True
+                break
+
+def print_help():
+    '''Print help about this program to console.'''
+    print(
+"""This script checks for errors in Minetest mod translation files
+(*.tr and template.txt) and prints any errors on the console.
+
+Syntax:
+    """+_argv[0]+""" <PATH> [OPTIONS]
+
+<PATH> is the path to check (default: working directory).
+OPTIONS is a list of optional options. These options are supported:
+
+    -h, --help: Print help and exits program
+    -e, --list-empty: List files with empty translations""")
+
+def main():
+    '''Starting point of the program.'''
+    # Scan parameters
+    set_params(_argv)
+    toptions = {
+        "list_empty": False,
+    }
+    if params["help"]:
+        print_help()
+        return
+    if params["list-empty"]:
+        toptions["list_empty"] = True
+    # Run tests
+    path = "."
+    if len(_argv) > 1:
+        path = _argv[1]
+    else:
+        print("No path specified.")
+        print("Use '"+_argv[0]+" --help' for help.")
+        return
+    run_tests(path, toptions)
+
+main()
```

Comment:
Can we make the script importable, i.e. only call the `main()` function if it's run on their own? This would make it easier to incorporate with other custom scripts, e.g. in a test suite.

Example:

```suggestion
if __name__ == "__main__":
    main(_argv)
```